provider/cloudflare/cloudflare.go

```diff
@@ -339,7 +344,22 @@ func (p *CloudFlareProvider) submitChanges(ctx context.Context, changes []*cloud
 // AdjustEndpoints modifies the endpoints as needed by the specific provider
 func (p *CloudFlareProvider) AdjustEndpoints(endpoints []*endpoint.Endpoint) []*endpoint.Endpoint {
 	adjustedEndpoints := []*endpoint.Endpoint{}
+
+	zoneNameIDMapper, idToZoneMapper, err := p.prepareZoneMappings()
+	if err != nil {
+		log.WithFields(log.Fields{
+			"endpoints": endpoints,
+		}).WithError(err).Errorf("Cannot adjust endpoints because zone mapping failed")
+
+		return endpoints
```

Comment:
I think we should return the error here, because something went wrong in this function.

Reply:
That would require modifying the Provider Interface though and the rest of the `AdjustEndpoints` implementations

Reply:
This would fail if the `Zones` API request in the `prepareZoneMappings` functions failed, which means that there is an issue with the Cloudflare API and not ExternalDNS itself. That's why I went with logging the error

Reply:
@Raffo What do you think about this?

Reply:
@Raffo Bump

Reply:
That makes sense, thanks.